### PR TITLE
Rewrite audit report trail

### LINKED ISSUE #761: Re - Write AuditTrailReport In the new React UI
Link : https://oe3.openelis-global.org/api/OpenELIS-Global/AuditTrailReport

### LINKED ISSUE #767: Average Turn Around Time returns NaN
1. Section: Dashboard -> Average Turn Around time

2. The Average Turn Around time container on the dashboard shows "NaN" instead of 0 if there's no value. The expanded view too shows NaN. The data being returned by the backend is 
`{receptionToResult: 'NaN', resultToValidation: 'NaN', receptionToValidation: 'NaN'}`

3. ### Supposed Fix

This can be fixed in two ways.

- Either it's fixed from backend by returning 0 or frontend uses conditional to render 0

4. Attach relevant screenshots and video recordings if necessary.
<br/>

image

<br/>

image

<br/>

5. Indicate the severity of the issue using the following ratings:
    1. This issue is something we would like to see improved in the future but does not significantly affect our work.

cc. @mozzy11 @CalebSLane

### PR DESCRIPTION
**Pull request by Lovelyfin00**
Fixes #761 

## Fix Summary

- [x] Migrated the `Audit Trail report` from JSP to new React UI
- [x] Added breadcrumbs

## Screen recording

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/38161296/c4d60fbc-8a45-421a-b3e1-f634854ecd7b

### CODE REVIEW

File: frontend/src/components/Reports/Routine.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ export const RoutineReportsMenu = {
       icon: IbmWatsonNaturalLanguageUnderstanding,
       SideNavMenuItem: [
         {
-          link: config.serverBaseUrl + "/AuditTrailReport",
+          link: "/RoutineReport/AuditTrailReport",
```
Comment (manishjha-04) [inline frontend/src/components/Reports/Routine.js:132]: Here it should be `link: "/AuditTrailReport",` just same as you have done for study.js below

Comment (Lovelyfin00) [reply frontend/src/components/Reports/Routine.js:132]: Hi @manishjha-04 
I did that but was still getting redirected to `ttps://localhost/api/OpenELIS-Global/AuditTrailReport`

Comment (manishjha-04) [reply frontend/src/components/Reports/Routine.js:132]: Have you created route in app.js for  /AuditTrailReport ,that i asked you for?

Comment (Lovelyfin00) [reply frontend/src/components/Reports/Routine.js:132]: Yes
I just did it now and it works
Thanks a bunch

File: frontend/src/components/Reports/auditTrailReport/AuditTrailReport.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                    />
+                </Button>
+              </Section>
+            </Column>
+          </Grid>
+        </Form>
+      </>
+    )
+}
+
```
Comment (sherrif10) [inline frontend/src/components/Reports/auditTrailReport/AuditTrailReport.js:102]: Resolve merge conflicts as well. @Lovelyfin00  well done.

Comment (Lovelyfin00) [reply frontend/src/components/Reports/auditTrailReport/AuditTrailReport.js]: fixed

**Comment by Lovelyfin00** [comment]
Hi @mozzy11 @manishjha-04 @sherrif10 @CalebSLane 

This PR is ready for review now
Thanks a bunch😅😅

File: frontend/src/components/Reports/auditTrailReport/AuditTrail.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        window.open(url);
+        setIsLoading(false);
+        setShowNotification(true);
+    }
+
+    return (
+        <>
+        <br />
+        <Grid fullWidth={true}>
```
Comment (sherrif10) [inline frontend/src/components/Reports/auditTrailReport/AuditTrail.js:46]: Well done. Do you mind sharing the final vedio please.

**Review by sherrif10 — APPROVED** [review]

File: frontend/src/components/Reports/auditTrailReport/AuditTrail.js (hunk 3 above)
Comment (Lovelyfin00) [reply frontend/src/components/Reports/auditTrailReport/AuditTrail.js:46]: Hi @sherrif10 
The vidoe is what I included in the description above

**Review by mozzy11 — CHANGES_REQUESTED** [review]
hello @Lovelyfin00 , Nice progress. 
Tis will require rewriting the actual report page in React

**Comment by Lovelyfin00** [comment]
> hello @Lovelyfin00 , Nice progress. Tis will require rewriting the actual report page in React

Hi @mozzy11 

Yeah I did that. You can checkout the video I included in the description. When ut comes to making the request, I called the endpoint but got an html response and not a json. So there was no way for me to display it which is why i had to redirect to the url

**Comment by mozzy11** [comment]
yeah , 
youll need to re-write this [controller](https://github.com/I-TECH-UW/OpenELIS-Global-2/blob/develop_3x/src/main/java/org/openelisglobal/audittrail/controller/AuditTrailReportController.java)  as a REST controller

**Comment by Lovelyfin00** [comment]
> yeah , youll need to re-write this [controller](https://github.com/I-TECH-UW/OpenELIS-Global-2/blob/develop_3x/src/main/java/org/openelisglobal/audittrail/controller/AuditTrailReportController.java) as a REST controller

That makes alot of sense, thanks. Is there a way to get a correct accessionNumber @mozzy11

**Comment by Lovelyfin00** [comment]
> @Lovelyfin00 what do you mean to get the correct correct accessionNumber ??

I actually meant Lab No
When i search using Lab No, I don't know how the UI displays the result because I don't have a correct lab No to search for

**Comment by Lovelyfin00** [comment]
Hi @mozzy11 
I don't know Java so I might not be able to writeout the code for the rest controller

File: frontend/src/App.js [hunk 4]
```diff
@@ -378,6 +380,12 @@ export default function App() {
                   component={() => <RoutineReports />}
                   role="Reports"
                 />
+                <SecureRoute
+                  path="/AuditTrailReport"
```
Comment (mozzy11) [inline frontend/src/App.js:384]: use the existing Route for `/Routine`

File: frontend/src/components/Reports/Routine.js [hunk 5]
```diff
@@ -123,8 +123,8 @@ export const RoutineReportsMenu = {
       icon: IbmWatsonNaturalLanguageUnderstanding,
       SideNavMenuItem: [
         {
-          link: config.serverBaseUrl + "/AuditTrailReport",
-          label: <FormattedMessage id="sideNav.label.audittrail" />,
+          link: "/AuditTrailReport?type=routine",
```
Comment (mozzy11) [inline frontend/src/components/Reports/Routine.js:126]: use    `link: "/Routinetype=routine", and render the report from [here  ](https://github.com/I-TECH-UW/OpenELIS-Global-2/blob/develop_3x/frontend/src/components/Reports/routine/Index.js#L44-L104)

**Comment by mozzy11** [comment]
Hello @Lovelyfin00 , addres the above comments , i can merge this in and Have some one who can complete the API

**Comment by Lovelyfin00** [comment]
Hi @mozzy11 
I have made the requested changes. Here is a video demo of them
<br/>


https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/38161296/d058e607-abaa-4b38-b1e0-b0faf457c511

<br/>

I keep seeing this error even though I have added the id to both translation files.
<br/>
<hr/>

image

**Review by mozzy11 — APPROVED** [review]

File: frontend/src/languages/fr.json [hunk 6]
```diff
@@ -20,21 +21,18 @@
   "from.title": "De",
   "to.title": "À",
   "report.label.site.dateType": "Type de date",
-  "banner.menu.reports": "Rapports",
```
Comment (parthnagdev) [inline frontend/src/languages/fr.json:23]: hey @Lovelyfin00 why was this line removed? what issue was it causing?

Comment (Lovelyfin00) [reply frontend/src/languages/fr.json:23]: @parthnagdev 
It was duplicated
Removed it because it showed up twice on the file

Comment (Lovelyfin00) [reply frontend/src/languages/fr.json:23]: @parthnagdev 
Sorry, I just cross checked
I thought it was duplicated as a result of merge conflicts.
I can add it back right now

Comment (Lovelyfin00) [reply frontend/src/languages/fr.json:23]: I opened a [PR for this ](https://github.com/I-TECH-UW/OpenELIS-Global-2/pull/886)
